doc/nrf/app_boards.rst

```diff
@@ -0,0 +1,102 @@
+.. _app_boards:
+
+Board support
+#############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |NCS| provides board definitions for all Nordic Semiconductor devices.
+In addition, you can define custom boards.
+
+.. _gs_programming_board_names:
+
+Board names
+***********
+
+The following tables lists all boards and build targets for Nordic Semiconductor's hardware platforms.
```

Comment:
```suggestion
The following table lists all boards and build targets for Nordic Semiconductor's hardware platforms.
```
We need to ensure that we are always informed when a new build target is added, I understand this is not the case right now.